api/src/opentrons/protocol_api/protocol_context.py

```diff
@@ -1024,17 +1046,33 @@ def deck(self) -> Deck:
 
     @property  # type: ignore
     @requires_version(2, 0)
-    def fixed_trash(self) -> Labware:
+    def fixed_trash(self) -> Union[Labware, TrashBin]:
         """The trash fixed to slot 12 of the robot deck.
 
-        It has one well and should be accessed like labware in your protocol.
+        In API Versions prior to 2.16 it has one well and should be accessed like labware in your protocol.
         e.g. ``protocol.fixed_trash['A1']``
+
+        In API Version 2.16 and above it returns a Trash fixture for OT-2 Protocols.
         """
+        if self._api_version >= APIVersion(2, 16):
+            if self._core.robot_type == "OT-3 Standard":
+                raise APIVersionError(
+                    "Fixed Trash is not supported on Flex protocols in API Version 2.16 and above."
+                )
```

Comment:
Let's raise `NoTrashDefinedError` instead of `APIVersionError`. Whether `protocol.fixed_trash` is available is a question of more than just the `apiLevel`; it has to do with the robot type, too.

```suggestion
                raise NoTrashDefinedError(
                    "Fixed Trash is not supported on Flex protocols in API Version 2.16 and above."
                )
```